command/agent/auth/auth.go

```diff
@@ -254,9 +255,22 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 		}
 
 		// This should only happen if there's no preloaded token (regular auto-auth login)
-		//  or if a preloaded token has expired and is now switching to auto-auth.
+		// or if a preloaded token has expired and is now switching to auto-auth.
 		if secret.Auth == nil {
-			secret, err = clientToUse.Logical().WriteWithContext(ctx, path, data)
+			isTokenFileMethod = path == "auth/token/lookup-self"
+			if isTokenFileMethod {
+				token, _ := data["token"].(string)
+				lookupSelfClient, err := clientToUse.Clone()
+				if err != nil {
+					ah.logger.Error("failed to clone client to perform token lookup")
+					return err
+				}
+				lookupSelfClient.SetToken(token)
+				secret, err = lookupSelfClient.Auth().Token().LookupSelf()
```

Comment:
Maybe we should log whether the token is periodic, and if not, when the token will stop working / reach max ttl?

Reply:
Also since batch tokens can't be renewed, that's another case where we know in advance when Agent will stop working, and we should probably inform the user.

Reply:
Thanks! This makes a lot of sense. I admittedly didn't think much about batch tokens, but it should be easy enough to log when one is in use.